common/src/nonJvmMain/kotlin/DiscordBitSet.kt

```diff
@@ -2,14 +2,14 @@ package dev.kord.common
 
 import com.ionspin.kotlin.bignum.integer.BigInteger
 import com.ionspin.kotlin.bignum.integer.Sign
-import io.ktor.utils.io.core.*
+import kotlinx.io.Buffer
+import kotlinx.io.readByteArray
 
-internal actual fun formatIntegerFromLittleEndianLongArray(data: LongArray) =
-    withBuffer(data.size * Long.SIZE_BYTES) {
-        // need to convert from little-endian data to big-endian expected by BigInteger
-        writeFully(data.reversedArray())
-        BigInteger.fromByteArray(readBytes(), Sign.POSITIVE).toString()
-    }
+internal actual fun formatIntegerFromLittleEndianLongArray(data: LongArray): String {
+    val buffer = Buffer()
+    data.reversedArray().forEach(buffer::writeLong)
+    return BigInteger.fromByteArray(buffer.readByteArray(), Sign.POSITIVE).toString()
+}
```

Comment:
just in case, an alternative implementation:
```kotlin
internal actual fun formatIntegerFromLittleEndianLongArray(data: LongArray): String {
    // need to convert from little-endian data to big-endian expected by BigInteger
    val bytes = ByteArray(size = data.size * Long.SIZE_BYTES)
    val lastIndex = data.lastIndex
    for (i in 0..lastIndex) {
        val long = data[i]
        val offset = (lastIndex - i) * Long.SIZE_BYTES
        bytes[offset + 0] = long.ushr(56).toByte()
        bytes[offset + 1] = long.ushr(48).toByte()
        bytes[offset + 2] = long.ushr(40).toByte()
        bytes[offset + 3] = long.ushr(32).toByte()
        bytes[offset + 4] = long.ushr(24).toByte()
        bytes[offset + 5] = long.ushr(16).toByte()
        bytes[offset + 6] = long.ushr(8).toByte()
        bytes[offset + 7] = long.ushr(0).toByte()
    }
    return BigInteger.fromByteArray(bytes, Sign.POSITIVE).toString()
}
```